samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/BookingDialog.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.corebot.app.insights;
+
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.dialogs.DialogTurnResult;
+import com.microsoft.bot.dialogs.WaterfallDialog;
+import com.microsoft.bot.dialogs.WaterfallStep;
+import com.microsoft.bot.dialogs.WaterfallStepContext;
+import com.microsoft.bot.dialogs.prompts.ConfirmPrompt;
+import com.microsoft.bot.dialogs.prompts.PromptOptions;
+import com.microsoft.bot.dialogs.prompts.TextPrompt;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.InputHints;
+import com.microsoft.recognizers.datatypes.timex.expression.Constants;
+import com.microsoft.recognizers.datatypes.timex.expression.TimexProperty;
+
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The class containing the booking dialogs.
+ */
+public class BookingDialog extends CancelAndHelpDialog {
+    private final String destinationStepMsgText = "Where would you like to travel to?";
+    private final String originStepMsgText = "Where are you traveling from?";
+    
+    public BookingDialog() {
+        super("BookingDialog");
+        
+        addDialog(new TextPrompt("TextPrompt"));
+        addDialog(new ConfirmPrompt("ConfirmPrompt"));
+        addDialog(new DateResolverDialog());
+        WaterfallStep[] waterfallSteps = {
+            this::destinationStep,
+            this::originStep,
+            this::travelDateStep,
+            this::confirmStep,
+            this::finalStep
+        };
+        
+        addDialog(new WaterfallDialog("WaterfallDialog", Arrays.asList(waterfallSteps)));
+    }
+    
+    
+    private CompletableFuture<DialogTurnResult> destinationStep(WaterfallStepContext stepContext) {
+        BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
+        
+        if (bookingDetails.getDestination == null) {
+            Activity promptMessage = MessageFactory.text(destinationStepMsgText, destinationStepMsgText);
+            PromptOptions promptOptions = new PromptOptions();
+            promptOptions.setPrompt(promptMessage);
+            return stepContext.prompt("TextPrompt", promptOptions);
+        }
+        
+        return stepContext.next(bookingDetails.getDestination);
+    }
+    
+    
+    private CompletableFuture<DialogTurnResult> originStep(WaterfallStepContext stepContext) {
+        BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
+
+        bookingDetails.setDestination(stepContext.getResult().toString());
```

Comment:
```suggestion
        bookingDetails.setDestination((String) stepContext.getResult());
```

I don't want to have any problem if `getResult` is a null object and the `toString()` will fail